payjoin/src/core/into_url.rs

```diff
@@ -25,6 +28,41 @@ impl From<ParseError> for Error {
 
 type Result<T> = core::result::Result<T, Error>;
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Url(pub(crate) url::Url);
+
+impl Url {
+    pub fn as_str(&self) -> &str { self.0.as_ref() }
+}
+
+impl From<Url> for url::Url {
+    fn from(value: Url) -> Self { value.0 }
+}
+
+impl TryFrom<&str> for Url {
+    type Error = Error;
+    fn try_from(s: &str) -> Result<Self> { Ok(Url(url::Url::parse(s)?)) }
+}
+
+impl TryFrom<String> for Url {
+    type Error = Error;
+    fn try_from(s: String) -> Result<Self> { Url::try_from(s.as_str()) }
+}
```

Comment:
These from methods allow us to still convert a `into_url::Url` to a `url::Url` for use in external crates.

Not sure if we really want that to be an approach to encourage over just using `url::Url::parse()` externally instead/

Reply:
I don't think we want `impl From<Url> for url::Url` because that's part of the public API and we're better off exposing our own functions.